### PR TITLE
[OOT] Re-enabled support for bootstrap token injection

### PR DESCRIPTION
**Pull request by zuzzas**
**What this PR does / why we need it**:

Bootstrap token injection stopped working for Machines that are managed by the new OOT Machine controller. I've re-enabled previous behavior.

**Release note**:

```improvement operator
Bootstrap token injection now works in the new OOT Machine controller
```

### CODE REVIEW

**Comment by prashanth26** [comment]
Hi zuzzas,

This is something even I realized off late, sorry for not fixing it yet. Was just about to create a fix for the same. Thanks for the PR 👍  Looks like a few unit tests are failing. Can you please re-check?

**Comment by zuzzas** [comment]
I've reproduced the issue locally (will fix it soon), but I'm having some problems with accessing tests results in the Concourse. I've logged in, of course.

image

**Comment by zuzzas** [comment]
@prashanth26 
Fixed! CI tests should pass too. I won't even wait, that's how determined I am! :)

**Review by prashanth26 — APPROVED** [review]
LGTM.

File: pkg/util/provider/machinecontroller/machine.go [hunk 1]
```diff
@@ -299,6 +299,14 @@ func (c *controller) triggerCreationFlow(createMachineRequest *driver.CreateMach
 		return retry, err
 	}
 
+	// we should avoid mutating Secret, since it goes all the way into the Informer's store
```
Comment (prashanth26) [inline pkg/util/provider/machinecontroller/machine.go:302]: 100%

**Pull request merged**